File: solutions/ruby/anagram/1/anagram.rb
class Anagram
  def initialize(str)
    @word = str.downcase
  end
  def match(words)
    words.filter{is_anagram?(_1)}
  end
  def is_anagram?(str)
    return false if str.downcase == @word
    letters = str.downcase.chars
    @word.chars.each do |chr|
      i = letters.find_index(chr)
      return false unless i
      letters.delete_at(i)
    end
    return letters.empty?
  end
end

File: solutions/python/anagram/1/anagram.py
def find_anagrams(word, candidates):
    def is_anagram(wrd2):
        wrd1, wrd2 = word.lower(), wrd2.lower()
        if wrd1 == wrd2 or len(wrd1) != len(wrd2):
            return False
        wrdlist = list(wrd2)
        for chr in wrd1:
            if chr in wrdlist:
                wrdlist[wrdlist.index(chr)] = None
            else:
                return False
        return True

    return filter(is_anagram, candidates)
